<launch>
  <arg name="map_publish_rate" default="20" />
  <arg name="polygon_simplify_tolerance" default="0.1" />

  <arg name="enable_radar_cluster" default="False" />
  <arg name="enable_lidar_cluster" default="True" />
  <arg name="enable_vision_cluster" default="True" />
  <arg name="enable_raw_lidar_points" default="False" />
  <arg name="enable_raw_radar_points" default="True" />
  <arg name="enable_lane_marker" default="True" />

  <arg name="lidar_z_min" default="-1.5" />
  <arg name="lidar_z_max" default="1.0" />
  <arg name="lidar_shape_radius" default="0.15" />
  <arg name="lidar_priority" default="0.25" />
  <arg name="lidar_discard_probability" default="0.9" />

  <arg name="radar_z_min" default="0.2" />
  <arg name="radar_z_max" default="3.0" />
  <arg name="radar_shape_radius" default="0.15" />
  <arg name="radar_priority" default="0.25" />
  <arg name="radar_discard_probability" default="0.9" />

  <arg name="enable_merge_filter" default="True" />
  <arg name="merge_growth_distance" default="0.65" />
  <arg name="min_merging_overlap_percent" default="0.5" />
  <arg name="min_merging_overlap_area" default="1.0" />

  <arg name="enable_lane_index_filter" default="True" />

 <!--
  <node pkg="mapping" type="mapping_data_integration.py" name="mapping_data_integration" output="screen">
    <param name="map_publish_rate" value="$(arg map_publish_rate)" />
    <param name="polygon_simplify_tolerance" value="$(arg polygon_simplify_tolerance)" />
    <param name="enable_radar_cluster" value="$(arg enable_radar_cluster)" />
    <param name="enable_lidar_cluster" value="$(arg enable_lidar_cluster)" />
    <param name="enable_vision_cluster" value="$(arg enable_vision_cluster)" />
    <param name="enable_raw_lidar_points" value="$(arg enable_raw_lidar_points)" />
    <param name="enable_raw_radar_points" value="$(arg enable_raw_lidar_points)" />
    <param name="enable_lane_marker" value="$(arg enable_lane_marker)" />

    <param name="lidar_z_min" value="$(arg lidar_z_min)" />
    <param name="lidar_z_max" value="$(arg lidar_z_max)" />
    <param name="lidar_shape_radius" value="$(arg lidar_shape_radius)" />
    <param name="lidar_priority" value="$(arg lidar_priority)" />
    <param name="lidar_discard_probability" value="$(arg lidar_discard_probability)" />

    <param name="enable_merge_filter" value="$(arg enable_merge_filter)" />
    <param name="merge_growth_distance" value="$(arg merge_growth_distance)" />
    <param name="min_merging_overlap_percent" value="$(arg min_merging_overlap_percent)" />
    <param name="min_merging_overlap_area" value="$(arg min_merging_overlap_area)" />
    <param name="enable_lane_index_filter" value="$(arg enable_lane_index_filter)" />
  </node>
  -->
  <node pkg="mapping" type="debug_wrapper.py" name="mapping_data_integration" output="screen" args="--debug_node=mapping_data_integration.py --debug_port=53000 --debug_wait">
    <param name="map_publish_rate" value="$(arg map_publish_rate)" />
    <param name="polygon_simplify_tolerance" value="$(arg polygon_simplify_tolerance)" />
    <param name="enable_radar_cluster" value="$(arg enable_radar_cluster)" />
    <param name="enable_lidar_cluster" value="$(arg enable_lidar_cluster)" />
    <param name="enable_vision_cluster" value="$(arg enable_vision_cluster)" />
    <param name="enable_raw_lidar_points" value="$(arg enable_raw_lidar_points)" />
    <param name="enable_raw_radar_points" value="$(arg enable_raw_lidar_points)" />
    <param name="enable_lane_marker" value="$(arg enable_lane_marker)" />

    <!-- Raw Lidar entity configuration -->
    <param name="lidar_z_min" value="$(arg lidar_z_min)" />
    <param name="lidar_z_max" value="$(arg lidar_z_max)" />
    <param name="lidar_shape_radius" value="$(arg lidar_shape_radius)" />
    <param name="lidar_priority" value="$(arg lidar_priority)" />
    <param name="lidar_discard_probability" value="$(arg lidar_discard_probability)" />

    <!-- Map filter configuration -->
    <param name="enable_merge_filter" value="$(arg enable_merge_filter)" />
    <param name="merge_growth_distance" value="$(arg merge_growth_distance)" />
    <param name="min_merging_overlap_percent" value="$(arg min_merging_overlap_percent)" />
    <param name="min_merging_overlap_area" value="$(arg min_merging_overlap_area)" />
    <param name="enable_lane_index_filter" value="$(arg enable_lane_index_filter)" />
  </node>

</launch>
